svAssistant.py

```diff
@@ -22,97 +23,42 @@
-"Hops": {"grow_time": 11, "produce_time": 1, "harvests/crop": 999, "yield": 1},
-"Hot Pepper": {"grow_time": 5, "produce_time": 3, "harvests/crop": 999, \
-                                                                "yield": 1},
-"Melon": {"grow_time": 12, "harvests/crop": 1, "yield": 1},
-"Radish": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Red Cabbage": {"grow_time": 9, "harvests/crop": 1, "yield": 1},
-"Starfruit": {"grow_time": 13, "harvests/crop": 1, "yield": 1},
-"Tomato": {"grow_time": 11, "produce_time": 4, "harvests/crop": 5, \
-                                                                "yield": 1},
-"Wheat": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Poppy": {"grow_time": 7, "harvests/crop": 1, "yield": 1},
-"Summer Spangle": {"grow_time": 8, "harvests/crop": 1, "yield": 1},
-"Amaranth": {"grow_time": 7, "harvests/crop": 3, "yield": 1},
-"Artichoke": {"grow_time": 8, "harvests/crop": 1, "yield": 1},
-"Beet": {"grow_time": 6, "harvests/crop": 1, "yield": 1},
-"Bok Choy": {"grow_time": 4, "harvests/crop": 1, "yield": 1},
-"Cranberries": {"grow_time": 7, "produce_time": 5, "harvests/crop": 5, \
-                                                                "yield": 2},
-"Eggplant": {"grow_time": 5, "produce_time": 5, "harvests/crop": 999, \
-                                                                "yield": 1},
-"Grape": {"grow_time": 10, "produce_time": 3, "harvests/crop": 999, \
-                                                                "yield": 1},
-"Pumpkin": {"grow_time": 13, "harvests/crop": 1, "yield": 1},
-"Yam": {"grow_time": 10, "harvests/crop": 1, "yield": 1},
-"Fairy Rose": {"grow_time": 12, "harvests/crop": 1, "yield": 1},
-"Sunflower": {"grow_time": 8, "harvests/crop": 1, "yield": 1}
-}
+# Creates a dictionary containing the buy and sell price of crops. Retrieves
+# values from crops_store_values.csv
+crops_store_values = {}
+with open('crops_store_values.csv') as crop_values:
+    read_csv = csv.reader(crop_values, delimiter=',')
+
+    for row in read_csv:
+        if len(row) == 0 or row[0][0] == "#":
+            continue
+        else:
+            crops_store_values[row[0]] = {"buy": float(row[1]), 'sell': \
+                                                                float(row[2])}
+
+# Creates a dictionary containing the grow_time, produce time (if applicable),
+# harvests per crop, and yield of each crop. Retrieves the values from 
+# crops_growth_values.csv
+crops_growth_values = {}
+with open('crops_growth_values.csv') as crop_growth:
+    read_csv = csv.reader(crop_growth, delimiter=',')
+
+    for row in read_csv:
+        # Tells it to ignore empty lines and lines starting with # (to allow 
+        # for commenting on the .csv file)
+        if len(row) == 0 or row[0][0] == "#":
+            continue
+        elif row[0] in multiharvest_crops:
+            crops_growth_values[row[0]] = {"grow_time": float(row[1]), \
```

Comment:
A csv shouldn't have variable columns.  If a crop doesn't have produce_time then the field can be left blank with a ,, or set it to 0 for no produce time.  This simplifies your parsing here as well.

Also your multiharvest crops array could be generated by adding crops that have > 0 produce_time.